routes/ingestion.py

```
@@ -73,70 +72,72 @@ async def scan_all_channels():
+@router.post("/scan/{source_id}")
+async def scan_source(source_id: str):
+    """Trigger a scan of a specific source."""
     from app.services.channel_scanner import ChannelScanner
 
     try:
         scanner = ChannelScanner()
-        result = scanner.scan_channel_by_id(channel_id)
+        result = scanner.scan_channel_by_id(source_id)
         return {"status": "success", **result}
     except ValueError as e:
         raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)